### DISCUSSION TITLE
Micropython Expert help needed

### DISCUSSION THREAD

**Discussion by PYTRONIX-INC** [micropython/micropython#10262] [General]
hello,

We are a Canadian company based in Ottawa, ON. we are looking for an expert in micropython for a project. We have tried to look for micropython experts in other platforms but the skill level was not a good fit for our needs. 

Hope this is OK to post here as we are not getting anywhere with other platforms. anyone interested in doing a paid side project please contact me.

**Comment by scruss** [comment]
What are you looking for? Porting? Library development?

**Comment by PYTRONIX-INC** [comment]
Hello,

I need to port STM32H723ZGT6 to micropython

On Sun, Dec 18, 2022 at 3:15 PM Stewart Russell ***@***.***>
wrote:

> What are you looking for? Porting? Library development?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/micropython/discussions/10262#discussioncomment-4443303>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AXXGJRFELJEQR5P24GYGRJDWN5WEZANCNFSM6AAAAAATCCUXIQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***
> com>
>
-- 

Regards,



Slim  Ben Ghalba

*Phone & Whatsapp: *1-613-882-7645

*SkypeId:* slimbg99

*Pytronix Inc., *

*1000 Innovation Dr. Suite 500, Kanata, Ont. K2K 3E7*

*Canada*

**Reply by dpgeorge** [reply]
> I need to port STM32H723ZGT6 to micropython

This should be relatively straightforward because a lot of other H7 MCUs are already supported.

What level of porting do you need?  Is it simply the existing peripherals (GPIO, I2C, SPI), or do you need something additional?  Can you please list the features you need.

If it's a simple port I can do it as my role as a maintainer.  And I would ask for you to use GitHub sponsors to sponsor this project in return.

**Reply by PYTRONIX-INC** [reply]
hello Damien,

Really glad to hear from you. Pytronix Inc., will be more than glad to sponsor the project and make it available to the Micropython community. Pytronix Inc., is a power user of the micropython framework and we want to gladly contribute the community and spread the word about micropython.

With regards to the port, we are looking for the following features:
Built-in Ethernet Connectivity
Built-in SD Card
Built-in Direct-USB for Micropython  programming
8x UART
8 channel ADC
2x output DAC
16x GPIO (Timers included within GPIO)
1x I2C 
1x SPI

I am not familiar with the sponsor process, can you please send me the process.

**Reply by PYTRONIX-INC** [reply]
UPDATE: sponsoring done!

**Reply by dpgeorge** [reply]
Thank you very much for the sponsorship!

Those features should already be supported on the STM32H7xx series.  So it's just a matter of adding support for the particular STM32H723.

**Reply by dpgeorge** [reply]
See #10287 for initial STM32H723 support and a board definition for NUCLEO_H723ZG.  I do not yet have hardware so cannot test this.
